src/Microsoft.Data.Sqlite.Core/SqliteParameter.cs

```diff
@@ -141,7 +141,7 @@ public override string ParameterName
         /// Gets or sets the maximum size, in bytes, of the parameter.
         /// </summary>
         /// <value>The maximum size, in bytes, of the parameter.</value>
-        public override int Size { get; set; }
+        public override int Size { get; set; } = int.MaxValue;
```

Comment:
If you get the value before it's explicitly set, it should return the size of the `string` or `byte[]`. (Use an `int?` field to back it.) It can return `0` for other types.

Reply:
Okay, changed as suggested.